File: top_hot100/0104MaximumDepthofBinaryTree/max_depth.py
# coding: utf-8

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        return self.dfs(root)
    def dfs(self, root):
        if root == None:
            return 0
        num1 = self.dfs(root.left)
        num2 = self.dfs(root.right)
        return max(num1, num2) + 1
    

File: top_hot100/0104MaximumDepthofBinaryTree/max_depth.go
package maxdepth

// TreeNode 二叉树节点
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func maxDepth(root *TreeNode) int {
	return dfs(root)
}

func dfs(root *TreeNode) int {
	if root == nil {
		return 0
	}
	num1 := dfs(root.Left)
	num2 := dfs(root.Right)
	return max(num1, num2) + 1
}

func max(num1, num2 int) int {
	if num1 > num2 {
		return num1
	}
	return num2
}
